internal/reconcile_components.go

```diff
@@ -0,0 +1,126 @@
+package internal
+
+import (
+	"fmt"
+	"github.com/broadinstitute/terra-status-manager/internal/shared"
+	"github.com/broadinstitute/terra-status-manager/internal/statuspage"
+	"github.com/broadinstitute/terra-status-manager/pkg"
+	"github.com/go-resty/resty/v2"
+	"github.com/mitchellh/mapstructure"
+	"reflect"
+)
+
+// componentsToDelete provides a slice of remote components that don't correlate to an entry in the config
+func componentsToDelete(
+	configComponentMap map[string]pkg.Component,
+	remoteComponentMap map[string]statuspage.Component,
+) []statuspage.Component {
+	var ret []statuspage.Component
+	for name, remoteComponent := range remoteComponentMap {
+		if _, found := configComponentMap[name]; !found {
+			ret = append(ret, remoteComponent)
+		}
+	}
+	return ret
+}
+
+// componentsToCreate provides a slice of components that aren't present on the remote
+func componentsToCreate(
+	configComponentMap map[string]pkg.Component,
+	remoteComponentMap map[string]statuspage.Component,
+) ([]statuspage.Component, error) {
+	var ret []statuspage.Component
+	for name, configComponent := range configComponentMap {
+		if _, found := remoteComponentMap[name]; !found {
+			var c statuspage.Component
```

Comment:
Renamed, lemme know what you think. I tend to use `ret` when A) the variable is explicitly typed and B) the best name for the variable is the same as the function itself (I want to avoid that sort of lexical overloading).

I made the function names a bit more verbose to try to avoid that overloading, let me know your thoughts.